### PR TITLE
[Merged by Bors] - feat: two lemmas about tuple concatenation

### PR DESCRIPTION
**Pull request by eric-wieser**
Paired with https://github.com/leanprover-community/mathlib/pull/18192

---

- [x] depends on: #1637

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#1637
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1637~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Data/List/OfFn.lean [hunk 1]
```diff
@@ -4,7 +4,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Mario Carneiro
 
 ! This file was ported from Lean 3 source module data.list.of_fn
-! leanprover-community/mathlib commit 9003f28797c0664a49e4179487267c494477d853
+! leanprover-community/mathlib commit 4a578e41e86806aae7e84e8b936f8c976cfcce8e
```
[the remark this answers is not in the document]
Comment (eric-wieser) [reply Mathlib/Data/List/OfFn.lean:7]: Whoops, that's a blob SHA

**Comment by fpvandoorn** [comment]
bors merge

**Pull request closed without merging**